samples/build-duration-logger/index.js

```diff
@@ -0,0 +1,255 @@
+                // when the end of the stream is reached so use that as a rough proxy for the end of the stream
+                this.finishedProcessingBuild();
+
+                // Call the 'complete()' method on any handler that has it.
+                if (buildEventHandlers.complete) {
+                    buildEventHandlers.complete.forEach(handler => handler.complete());
+                }
+            },
+            eventListeners: [
+                {
+                    eventName: 'BuildEvent',
+                    eventHandler:event => {
+                        const buildEventPayload = JSON.parse(event.data);
+                        const { eventType } = buildEventPayload.type;
+
+                        if (this.allHandledEventTypes.has(eventType)) {
+                            buildEventHandlers[eventType].forEach(handler => handler[`on${eventType}`](buildEventPayload));
+                        }
+                    }
+                }
+            ],
+            retry: {
+                interval: 2000,
+                maxRetries: 100
+            }
+        });
+    }
+
+    finishedProcessingBuild() {
+        this.numBuildsInProcess--;
+        setTimeout(() => this.processPendingBuilds(), 0); // process the next set of pending builds, if any
+    }
+}
+
+function createServerSideEventStream(url, configuration) {
+    const STATUS_COMPLETE = 204;
+
+    let stream;
+    let retries;
+
+    const _onopen = configuration.onopen || noop;
+    const _onerror = configuration.onerror || noop;
+    const _oncomplete = configuration.oncomplete || noop;
+    const _configurationRetry = configuration.retry || { }
+    const _maxRetries = _configurationRetry.maxRetries || 0;
+    const _retryInterval = _configurationRetry.interval || 1000;
+
+    stream = createStream()
+
+    function createStream() {
+        stream = new EventSourcePolyfill(url, { headers: {'Authorization': `Basic ${BASIC_AUTH_TOKEN}`}});
+
+        stream.reconnectInterval = _retryInterval;
+
+        stream.onopen = (event) => {
+            retries = 0;
+            _onopen(event)
+        }
+
+        stream.onerror = (event) => {
+            // the server will sent a 204 status when closing the stream if no more content is present but the
+            // EventSource implementation handles it as an error, we map here from the error to the oncomplete
+            // callback as an improved API
+            if(event.status === STATUS_COMPLETE) {
+                _oncomplete()
+                return
+            }
+
+            // on errors (except STATUS_COMPLETE) EventSourcePolyfill will try to reconnect to the server until
+            // successfully unless we manually close it when _maxRetries is reached
+            if(stream.readyState === EventSourcePolyfill.CONNECTING) {
+                if(_maxRetries > 0 && retries < _maxRetries) {
+                    console.log(`Connecting to ${url}`);
```

Comment:
I'd maybe remove this entry as it's quite verbose, or add it as debug level output.

Reply:
Removed.